.rubocop.yml

```diff
@@ -49,6 +49,12 @@ RSpec/DescribeClass:
 RSpec/ExampleLength:
   Enabled: false
 
+# FIXME: Find a better way to block output during specs, and fix relevant examples.
+RSpec/ExpectOutput:
```

Comment:
Ah I see. This is a bit ugly but you could do something like this:

```ruby

shared_examples_for 'handling and recording the error' do
  def error
    expect { return src.syntax_tree }.to output.to_stderr
  end

  it 'does not raise an error' do
    expect { error }.to_not raise_error
  end

  it 'returns an empty syntax tree' do
    expect(error).to be_nil
  end

  it 'records the syntax error' do
    expect { src.syntax_tree }.to output(/#{error_class.name}/).to_stderr
  end

  it 'records the source name' do
    expect { src.syntax_tree }.to output(/#{source_name}/).to_stderr
  end

  it 'records the error message' do
    expect { src.syntax_tree }.to output(/#{error_message}/).to_stderr
  end
end
```

The `error` method takes advantage of the RSpec error suppression and returns the value so you can use it in a test